src/coreclr/src/jit/codegenarmarch.cpp

```diff
@@ -2658,6 +2658,26 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
                     INDEBUG_LDISASM_COMMA(sigInfo) nullptr, // addr
                     retSize MULTIREG_HAS_SECOND_GC_RET_ONLY_ARG(secondRetSize), ilOffset, target->GetRegNum());
     }
+#if defined(FEATURE_READYTORUN_COMPILER) && defined(TARGET_ARMARCH)
+    else if (call->IsR2RRelativeIndir())
+    {
+        // Generate a direct call to a non-virtual user defined or helper method
+        assert(callType == CT_HELPER || callType == CT_USER_FUNC);
+        assert(call->gtEntryPoint.accessType == IAT_PVALUE);
+
+        regNumber tmpReg = call->GetSingleTempReg();
```

Comment:
I might add an `assert(call->gtControlExpr == nullptr`);` here, and also add a comment saying that we're going to generate the call address into the temporary register that's been reserved.

Reply:
Sure. By the way I am thinking of porting the changes in this PR to the main one because it has all the details. I also need to update some of the other comments in this code.